### ISSUE TITLE
Intraface migration on upgrade

### ISSUE THREAD

**Issue by 2tata** [freifunk-gluon/gluon#2912]
**What is the problem?**

Oh upgrading the gluon firmware version from v2021.1.x(595abcf8cb2dc794801c6ed5f1641783fccf5806) to v2022.1.x(e9dcefee596fdc840ed23313286874879d4bc2d1) the migration scripts seems to ignore any custom interface configuration.

e.g. with VLANs
eth0.10 = WAN
eth0.20 = FF-Client
eth0.30 = FF-Mesh

the migrationsripts seems just appling the siteconfigs defauls

gluon.iface_single=interface
gluon.iface_single.name='/single'
gluon.iface_single.role='uplink'

**What is the expected behaviour?**

I assume that the migration scripts iterate over the bridge interfaces and take those interfaces listed in the bridges and apply them to the new network rohles.

**Comment by NeoRaider** [comment]
TL;DR: This migration is not implemented because it is difficult to do so correctly. We decided that we would only migrate what is officially supported (i.e. configurable though the Web UI), rather than attempting to do a migration that might cause additional breakage instead of helping.

---

There are a number of issues with migrating custom interfaces:
- Interfaces may change names between major OpenWrt releases (eth0 and eth1 switched for many devices in the ar71xx -> ath79 migration)
- Devices may be migrated from swconfig to DSA, splitting up what was a single interface into several

In fact, the update from v2021.1.x to v2022.1.x also lost all information on regular LAN/WAN interfaces if they had been modified in `/lib/gluon/core/sysconfig`. Unfortunately, we found no good way around this, as we don't even know which devices are affected. Even recovering Wired Mesh status for migration was somewhat fragile, because some devices changed from LAN+WAN to a single bridge interface, or vice-versa...

For regular LAN/WAN, we can now deal with changing interface names during upgrades, but custom interfaces may still break in the future, because they rely on referring to interfaces by name in the configuration. It would be nice to solve that at some point.
